src/components/CountryDetails.js

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect} from 'react';
+import { useParams } from 'react-router-dom';
+import { NavLink } from 'react-router-dom';
+import React from 'react';
+
+function CountryDetails (props) {
+
+  const [details, setDetails] = useState({});
+
+  const {countryId} = useParams();
+
+/*   Not sure why my page doesn't show anything at all on the browser. I have tried to apply the logic anyway to display the borders, but I have a feeling that my code is all over the place, sorry! */
+
+  useEffect(() => {
+      /* I should maybe have this in there somewhere? : country.alpha3Code === countryId; 
+    }); */
+    /* and something like this? : setDetails(prevValue) */
+  }, [countryId])
+
```

Comment:
to focus on the most important part I deleted all after return and wrote only the details name (details.name.common) and if I change the code to this, it's working:

```
import { useState, useEffect} from 'react';
import { useParams } from 'react-router-dom';
import React from 'react';

function CountryDetails (props) {

  const [details, setDetails] = useState(null); // J: I changed it to null

  const {countryId} = useParams();
  console.log(countryId)

/*   P: Not sure why my page doesn't show anything at all on the browser. I have tried to apply the logic anyway to display the borders, but I have a feeling that my code is all over the place, sorry! */

  useEffect(() => {
      /* P: I should maybe have this in there somewhere? : country.alpha3Code === countryId; 
    }); */


    // J: YES, you could for example write a message for that and invoke it here
    getDetails()

    /* P: and something like this? : setDetails(prevValue) */
    
  }, [countryId])

  const getDetails = ()=> {
    // J: use your props and do what you suggested, like this
    let country = props.allCountries.find((country)=> {
      return country.alpha3Code === countryId
    })
    console.log(country);
    setDetails(country)
  
  }

  // J: here we write a if condition to show loading, if details are an empty object

  if (details === null){
    return <p>Loading</p>
  }

    return (
        
        <h1>{details.name.common}</h1>
      
    )
}

export default CountryDetails;
```

Reply:
maybe it showed something at some point for you, but it was on the bottom of the page (see other comments to Routes)